flycheck.el

```diff
@@ -1753,6 +1789,20 @@ CHECKER supports buffers in the given major MODE.
 Return non-nil if CHECKER supports MODE and nil otherwise."
   (memq mode (flycheck-checker-get checker 'modes)))
 
+(defun flycheck-may-enable-checker (checker)
+  "Whether a generic CHECKER may be enabled for current buffer.
+
+Return non-nil if CHECKER may be used for the current buffer, and
+nil otherwise."
+  (if (member checker flycheck-enabled-checkers)
```

Comment:
Use `memq`, as we're dealing with symbols.  Also, try not to use `if`.  This can be rewritten as an `and` expression that's easier to follow, imho:

```elisp
(let ((enabled (flycheck-checker-get checker 'enabled))
      (shall-enable (and (not (flycheck-disabled-checker-p checker))
                         (or (memq checker flycheck-enabled-checkers)
                             (null enabled)
                             (funcall enabled checker)))))
  (if shall-enable
      (cl-pushnew checker flycheck-enabled-checkers)
    (cl-pushnew checker flycheck-disabled-checkers))
  shall-enable)
```

What do you think?